jung-algorithms/src/main/java/edu/uci/ics/jung/layout/spatial/Node.java

```diff
@@ -140,83 +158,39 @@ protected void split() {
     SE = new Node(x + width, y + height, width, height);
   }
 
-  public void visit(ForceObject<T> target) {
-    if (this.forceObject == null || target.getElement().equals(this.forceObject.getElement())) {
+  /**
+   * accept a visit from the visitor force object, and add this node's forces to the visitor
+   *
+   * @param visitor the visitor
+   */
+  public void acceptVisitor(ForceObject<T> visitor) {
+    if (this.forceObject == null || visitor.getElement().equals(this.forceObject.getElement())) {
```

Comment:
Are there ever circumstances under which someone might legitimately pass in a null `ForceObject`?  Or does that necessarily indicate a bug in our code?

If it indicates a bug, I'd probably add a Preconditions.checkNotNull() here.

Reply:
The visitor force object had better not be null and should be precondition checked, as you say.
the this.forceObject would be null in the case of an empty tree. That's not an error (as you probably knew)